### PR TITLE
hide progress indicator when connection refresh is done by codewhisperer

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Symptom

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/0a0a1272-1b22-4530-80f5-5fbe37b110f7

## After

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/e45f2caf-1184-49f4-97a3-a521b82e64ec

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/395c47af-0b98-419b-96e9-6f97f968753a

## Repro step
1. edit token file's `expiredAt` & `refreshToken` to make bearer connection expired and non-refreshable
2. keep typing or alt + c to manual trigger CW
3. since token is not refreshable, none of the service calls will succeed and plugin will recurrently retry with the broken refreshToken

https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/38ec1235-e56b-4d55-bfc0-3b55ebd97abc

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt [hunk 1]
```diff
@@ -105,9 +105,12 @@ class CodeWhispererService {
             return
         }
 
-        if (promptReAuth(project)) return
-
-        if (isQExpired(project)) return
+        if (isQExpired(project)) {
+            ApplicationManager.getApplication().executeOnPooledThread {
+                promptReAuth(project)
```
Comment (rli) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:110]: do we not need a return on this?

Comment (Will-ShaoHua) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:110]: yea, will update

Comment (andrewyuq) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:110]: as we discussed offline, return only if this reauth attempt fails so we don't fail this current request everytime.

Comment (Will-ShaoHua) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:111]: updated (and attached screen recording)

**Review by rli — APPROVED** [review]

**Pull request merged**